lemur/auth/views.py

```diff
@@ -616,6 +618,61 @@ def post(self):
         )
 
 
+class Vault(Resource):
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+
+        JWTAuthenticator.instance(
+            name="lemur_vault_authenticator",
+            audience=current_app.config.get("VAULT_CLIENT_ID"),
+            additional_issuers=[current_app.config.get("VAULT_ISSUER_URL"),],
+            timeout=1,
+            cache_to_file=True,)
+
+        super(Vault, self).__init__()
+
+    def get(self):
+        return "Redirecting..."
+
+    def post(self):
+        self.reqparse.add_argument("id_token", type=str, required=True, location="json")
+        args = self.reqparse.parse_args()
+        id_token = args["id_token"]
+        profile = JWTAuthenticator.instance("lemur_vault_authenticator").authenticate(id_token)
+
+        token_for_valid_audience = profile["aud"] == current_app.config.get("VAULT_CLIENT_ID")
+        user_in_authorized_group = False
+        for group in current_app.config.get("VAULT_AUTHORIZED_GROUPS"):
+            if group in profile['groups']:
+                user_in_authorized_group = True
+                break
+
+        if not token_for_valid_audience or not user_in_authorized_group:
+            return dict(message="The supplied credentials are invalid"), 403
+
+        user = user_service.get_by_email(profile["email"])
```

Comment:
There is one more immediate use-case that I think would be worth bringing in scope and supporting here. That would be authorizing users by e-mail address.

Reply:
(I can elaborate offline)